### ISSUE TITLE
Spring scheduling creating too many InProc dependencies

### ISSUE THREAD

**Issue by petergphillips** [microsoft/ApplicationInsights-Java#2870]
We recently introduced [Spring Cloud AWS](https://github.com/awspring/spring-cloud-aws) into our Spring Boot 3 application.

Shortly after releasing it to our development and pre-production environments we found that we had used up our Application Insights daily allowance with the `dependencies` that were generated by that application alone.

Investigating further it appears that these are all of type `InProc` with name `BatchingAcknowledgementProcessor$ScheduledAcknowledgementExecution$$Lambda$.run`.  According to https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#autocollect-inproc-dependencies-preview I didn't think that `InProc` were meant to be captured?

Adding in
```json
    "springScheduling": {
      "enabled": false
    },
```
to our application insights configuration stops the logs from filling up, but then also doesn't log any spring scheduling information.

#### Sample Application
https://github.com/ministryofjustice/hmpps-restricted-patients-api/tree/SDI-593-inproc-dependencies-2870 branch reproduces the issue.  Run with `dev,stdout,localstack` spring boot profiles enabled, setting the java agent to be applicationinsights-agent-3.4.7.jar and also set `APPLICATIONINSIGHTS_CONNECTION_STRING` and `APPLICATIONINSIGHTS_CONFIGURATION_FILE`.

### System information
 - SDK Version: 3.4.7, reproduced also on 3.4.8
 - OS type and version: Deployment using docker image eclipse-temurin:19-jre-jammy, locally reproduced on Mac OS X 12.5.1
 - Application Server type and version (if applicable): Spring Boot 3.0.1, also 3.0.2

 - Using spring-boot? Yes
 - Additional relevant libraries (with version, if applicable): Spring Cloud AWS v3.0.0-M3

### Logs
[applicationinsights.log](https://github.com/microsoft/ApplicationInsights-Java/files/10490928/applicationinsights.log)

**Comment by petergphillips** [comment]
for those that are also interested, I've also now worked out a better way of excluding the `InProc` dependencies, whilst still having logging of the rest of spring scheduling enabled:
```json
  "preview": {
    "sampling": {
      "overrides": [
        {
          "telemetryType": "dependency",
          "attributes": [
            {
              "key": "job.system",
              "value": "spring_scheduling",
              "matchType": "strict"
            },
            {
              "key": "code.namespace",
              "value": "io.awspring.cloud.sqs.listener.acknowledgement.BatchingAcknowledgementProcessor.*",
              "matchType": "regexp"
            }
          ],
          "percentage": 0
        }
      ]
    }
  }
```

**Comment by heyams** [comment]
@petergphillips I'm working on a fix upstream. will let you know when that fix becomes available. it's great to hear you have found a workaround in the meantime.

**Comment by trask** [comment]
The fix for this is released now in 3.4.10.

@petergphillips hopefully you shouldn't need the sampling override anymore in 3.4.10. thx!

**Comment by petergphillips** [comment]
great thanks, just had a go with 3.4.10 and no more inproc dependencies in app insights 👍
